homeassistant/components/modem_callerid/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Modem Caller ID integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from phone_modem import PhoneModem
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_DEVICE, DEFAULT_NAME, DOMAIN, EXCEPTIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"name": str, "device": str})
```

Comment:
We should provide a list of serial devices to choose from, instead of free text input. You can find an example in multiple integrations, most notably ZHA.